platforms/unix/plugins/SqueakSSL/sqUnixOpenSSL.c

```diff
@@ -95,25 +95,25 @@ enum sqMatchResult sqVerifyDNS(sqSSL* ssl, X509* cert, const char* serverName, c
 	return sqVerifyNameInner(ssl, cert, serverName, serverNameLength, GEN_DNS);
 }
 
-enum sqMatchResult sqVerifyNameInner(sqSSL* ssl, X509* cert, void* serverName, const size_t serverNameLength, const int matchType) {
+enum sqMatchResult sqVerifyNameInner(sqSSL* ssl, X509* cert, const void* serverName, const size_t serverNameLength, const int matchType) {
 	enum sqMatchResult matchFound = NO_MATCH_FOUND;
 
-	STACK_OF(GENERAL_NAME)* sANs = X509_get_ext_d2i(cert, NID_subject_alt_name, NULL, NULL);
+	STACK_OF(GENERAL_NAME)* sANs = sqo_X509_get_ext_d2i(cert, NID_subject_alt_name, NULL, NULL);
 	if (!sANs) {
 		if (ssl->loglevel) printf("sqVerifyNameInner: No sAN names\n");
 		matchFound = NO_SAN_PRESENT;
 	} else {
 		int i = 0;
-		int sANCount = sk_GENERAL_NAME_num(sANs);
+		int sANCount = sqo_sk_GENERAL_NAME_num(sANs);
 		for (i = 0; i < sANCount && matchFound != MATCH_FOUND; ++i) {
-			const GENERAL_NAME* sAN = sk_GENERAL_NAME_value(sANs, i);
+			const GENERAL_NAME* sAN = sqo_sk_GENERAL_NAME_value(sANs, i);
 			if ((sAN->type == matchType) &&
 			    sqVerifySAN(ssl, sAN, serverName, serverNameLength, matchType)) {
 				matchFound = MATCH_FOUND;
 				break;
 			}
 		}
-		sk_GENERAL_NAME_pop_free(sANs, GENERAL_NAME_free);
+		sqo_sk_GENERAL_NAME_pop_free(sANs, (void(*)(void*))sqo_sk_free);
```

Comment:
I don't understand what's going on here, but please double check if `(void(*)(void*))sqo_sk_free` is the correct substitution for `GENERAL_NAME_free` here.

Reply:
Because some expansion is taking place here that cannot be easily emulate. However, all `..._free` in the stack sense ultimately end up _being_ `sqo_sk_free`. Qt does it similarly.